### PR TITLE
record step as 'loading' if the loading screen is visible

### PR DESCRIPTION
**Pull request by roo2**
#### Changes proposed in this Pull Request
In https://github.com/Automattic/wp-calypso/issues/57494 I noticed that on the final loading screen in the signup hero flow, a `calypso_signup_step_start` event is sent with `step=intent`. But that is the same event as is sent by the original intent screen. I think this could be confusing, and it also could be useful to have an event for the loading screen.
The issue also affects the loading screen in the existing `onboarding` flow which currently sends a`calypso_signup_step_start` event with `step=domains`.

#### Testing instruction
Complete the `/signup?flags=signup/hero-flow` flow. On the first loading screen, you should see a tracks event `calypso_signup_step_start,step=loading,flow=onboarding`. On the second loading screen you should see a tracks event `calypso_signup_step_start,step=loading,flow=setup-site`.

EDIT FROM AUTUMN:   I think the above testing should be for  'calypso_signup_processing_screen_show'

### CODE REVIEW

**Comment by arthur791004** [comment]
Do you know why the step changed to the first one? I notice that but I don't dive into the problem. I think we should fix this problem. Then, we can track loading event when the value of `shouldShowLoadingScreen` changed from false to true or the actual component mounts (ex. ReskinnedProcessingScreen)

**Comment by roo2** [comment]
> Do you know why the step changed to the first one?

I think it's because the loading screen is not a real step, but rather just rendered if the `this.state.shouldShowLoadingScreen` flag is true. I'm not sure why the first step of the flow is used versus the last step, but since `loading` is not actually a step, I think we will need logic based on `this.state.shouldShowLoadingScreen`. (or we could make `loading` an actual step)

**Comment by arthur791004** [comment]
After diving into the problem, I found that when you submitted the last step, then
1. `goToNextStep` is triggered
2. `goToStep` is triggered. However, the `stepName` is undefined and `isEveryStepSubmitted` is false as the status doesn't update to `pending` immediately.
3. The URL changed to `/start/setup-site`
4. `redirectToFlow` is triggered. However, the current path name is invalid, so it redirects to the first step of the flow. Therefore, the step changed to the first step.

In short, I think we should fix this issue and then we can send the tracks event when `this.state.shouldShowLoadingScreen` changed to true. (But I'm not sure it would always show if the value is true) as I think changing the `stepName` to `loading` is a little weird. Or maybe the `stepName` becomes undefined? How do you think?

**Comment by autumnfjeld** [comment]
@roo2  could you also make sure the `intent` prop is included on the loading screen events?  This will help us to clearly differentiate if the loading screen is in the Write Flow or Build Flow.

As more "intent" flows are built out (DIFM, Importers) I imagine the intent prop will be become critical to analytics.

**Comment by roo2** [comment]
@arthur791004 hmm interesting your comment reminds me of some situations I saw while working on the "Skip this step" fix yesterday https://github.com/Automattic/wp-calypso/pull/. I don't understand exactly what you've seen but I wonder if that fix might change it?

**Comment by roo2** [comment]
> as I think changing the `stepName` to `loading` is a little weird

Hm maybe we should be sending a completely different event maybe `calypso_signup_processing_screen`?

**Comment by arthur791004** [comment]
>  hmm interesting your comment reminds me of some situations I saw while working on the "Skip this step" fix yesterday https://github.com/Automattic/wp-calypso/pull/. I don't understand exactly what you've seen but I wonder if that fix might change it?

I think it might not change it. What I've seen is that when the user submitted the last step, the status of that step won't change immediately. Then the signup framework continued to go to the next step but it is undefined (as it's the last step now). Therefore, the URL would change to the invalid path and then redirect to the first step of the flow.

> Hm maybe we should be sending a completely different event maybe calypso_signup_processing_screen?

Yeah, I think it might be better.

**Comment by roo2** [comment]
thanks @arthur791004 I updated to send a calypso_signup_processing_screen when `shouldShowLoadingScreen` changes to true and prevented it from sending the `calypso_signup_step_start` events while `shouldShowLoadingScreen` is true

**Review by arthur791004 — APPROVED** [review]
👍

File: client/lib/analytics/signup.js [hunk 1]
```diff
@@ -138,3 +138,20 @@ export function recordRegistration( { userData, flow, type } ) {
...
+/**
+ * Records loading of the processing screen
+ *
+ * @param {string} flow Signup flow name
+ * @param {string} previousStep The step before the processing screen
+ * @param {string} optionalProps Extra properties to record
+ */
+export function recordSignupProcessingScreen( flow, previousStep, optionalProps ) {
+	const device = resolveDeviceTypeByViewPort();
+	recordTracksEvent( 'calypso_signup_processing_screen', {
```
Comment (arthur791004) [inline client/lib/analytics/signup.js]: Oh maybe `calypso_signup_processing_screen_show` is better.

Comment (roo2) [reply client/lib/analytics/signup.js]: good point! I've updated that
